libnmstate/varlink/README.varlink.md

```diff
@@ -0,0 +1,164 @@
+# Varlink support for libnmstate
+Varlink protocol ([Varlink.org](https://varlink.org/)) can used to communicate with libnmstate via stdin/out and varlink client implementation ([Existing Language binding](https://varlink.org/Language-Bindings#how-to-test-new-language-bindings)). Varlink protocol encodes all messages as JSON object and communicates over unix and tcp socket connections. Nmstate service is defined in io.nmstate interface address and libnmstate functions should called in as specified in varlink interface defination file `io.nmstate.varlink`. This implementation supports python3-varlink version 29.0.0 and unix socket connection only.
+
+
+## Initiate nmstate varlink service
+Nmstate-varlink default listening address is ```unix:@nmstate``` and user can define the unix socket path using ```--address``` option. The current nmstate-varlink services limits only using unix sockets.
+
+unix socket connection
+```
+$ io.nmstate &
+
+# User defined socket address
+$ io.nmstate --address unix:/run/nmstate
+```
+
+systemd service initiation
+```
+$ systemctl start nmstate-varlink.service
+```
+After the `io.nmstate` interface is resolved using varlink resolver. Methods can be called directly via stdin/out without specifying the connection.
+`varlink call io.nmstate.Show`
+
+## Nmstate basic operations using varlink
+The basic functions from libnmstate (show, apply, commit and rollback) are called via varlink interface. Passing inputs to the functions only support JSON format. Below are the examples for each basic functions using varlink stdin/out and varlink client.
+ 
+Using libnmstate show function via varlink (query network state)
+
+Required for varlink stdin/out operation
+```
+$ dnf install libvarlink-utils
+```
+
+Varlink stdin/out:
+```
+$ varlink call unix:@nmstate/io.nmstate.Show
+
+# If interface is resolved using the varlink resolver
+$ varlink call io.nmstate.Show
+```
+Varlink python client:
+```python
+import varlink
+
+with varlink.Client("unix:@nmstate").open("io.nmstate") as nmstate:
+    nmstate.Show()
+```
+JSON output: libnmstate current network is reported under "state" object.
+```json
+{ 
+  "log": [
+    {
+      "level": "DEBUG",
+      "message": "Async action: Retrieve applied config: foo started",
+      "name": "root",
+      "time": "2020-08-05 10:22:19"
+    },
+    {
+      "level": "DEBUG",
+      "message": "Async action: Retrieve applied config: foo finished",
+      "name": "root",
```

Comment:
Can you elaborate on this `name: root` and why it is needed?

Reply:
@cathay4t 
I included same logging format parameters used nmstatectl. Should I remove it?

Reply:
Yes please, remove "name" parameter. IMO, it is not needed.